### ISSUE TITLE
Question...

### ISSUE THREAD

**Issue by bsbodden** [estruyf/vscode-remote-control#3]
I can see one can pass arguments like:

```
echo "{ \"command\": \"editorScroll\", \"args\": { \"to\": \"up\" } }"  | websocat ws://127.0.0.1:9999
```

The above command works, but the one below doesn't:

```
echo "{ \"command\": \"vscode.open\", \"args\": { \"uri\": \"file://some/file.py\" } }"  | websocat ws://127.0.0.1:9999 
```

The file in question is in the currently opened folder in VSCode. I don't know if it is a matter of escaping the URI. Is there a way to debug the extension? Thanks

**Comment by estruyf** [comment]
Hi @bsbodden, this requires an update on the extension and in your command.

**Extension code**

The `vscode.open` command expects a valid URI which needs to be parsed in code. Which is now implemented in version `v1.2.0`.

**Command**

The `vscode.open` command arguments need to be passed as follows:

```bash
echo "{ \"command\": \"vscode.open\", \"args\": [ \"file:///some/file.py\" ] }"  | websocat ws://127.0.0.1:9999
```

Instead of the object, you need to pass it as an array of arguments. This is what the command expects. The first argument is the URI, second is the options object. 

With the `editorScroll` command, this is different, as there it expects an object.

New version is out.

**Comment by bsbodden** [comment]
Thank you very much, Elio!
